packages/toolkit/src/createAsyncThunk.ts

```diff
@@ -617,7 +617,7 @@ export const createAsyncThunk = /* @__PURE__ */ (() => {
               }
               abortController.signal.addEventListener('abort', abortHandler)
             })
-            dispatch(
+            ;(dispatch as ThunkDispatch<unknown, unknown, UnknownAction>)(
```

Comment:
weird, i wonder what made this break

Reply:
It could be related to [this](https://github.com/microsoft/TypeScript/pull/56004). The problem is the second overload for `ThunkDispatch` currently looks like this:

```ts
<Action extends BasicAction>(action: Action): Action;
```

When it should look like this:

```ts
<Action extends BasicAction>(action: Action, ...extraArgs: any[]): Action;
```

The latter matches the actual type of `Dispatch` but the former does not. Obviosuly I don't know the types as well as you 
do but I think that might be the problem.

Reply:
honestly i always found it weird the types for Dispatch allowed for extra arguments when middleware and enhancers only ever pass through one

Reply:
I think [this](https://github.com/reduxjs/redux/pull/2560) might be somewhat relevant, if you want I can fix the type for `Dispatch` to not include extra arguments.